In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('df_2016.csv')
df.sample(3)
df.shape

(13051, 12)

# Pandas IV

1 - Cargad el csv que guardasteis ayer.
Antes de seguir con las preguntas, para evitar posibles errores con los nulos vamos a eliminar todos los nulos del DataFrame, esto lo haremos sobre el archivo de 2016.xlsx. Para ello vamos a utilizar el método .dropna() de Pandas. 

Tendréis que ejecutar la siguiente línea de código para eliminar los nulos del DataFrame y poder seguir haciendo el resto de los ejercicios.
```
df.dropna(inplace = True)
```
¿Cuántas filas se han eliminado con esta línea de código?

In [3]:
df.dropna(inplace=True)
df.shape

(12901, 12)

In [4]:
df.sample(10)

id_ticket       fecha  id_empleado  id_agente      categoría  \
2604   GMLENR-3142399065  2016-01-30        976.0       10.0  Acceso&%Login   
6328   KHLTST-5742725399  2016-12-21       1809.0       33.0        Sistema   
1921   GHLTNT-2042415734  2016-02-15        743.0       47.0  Acceso&%Login   
11265  TDRTER-1342573712  2016-07-22        971.0       47.0       Software   
489    GDLTNR-0042456154  2016-03-27       1305.0       41.0  Acceso&%Login   
9213   SHLTSR-1342627714  2016-09-14        391.0        7.0        Sistema   
658    GDLTNR-8042715769  2016-12-11       1036.0       27.0  Acceso&%Login   
378    GDLTER-6942545896  2016-06-24       1029.0       48.0       Software   
6358   KHLTST-7842722049  2016-12-18        234.0       20.0        Sistema   
5127   KHLEET-7542563187  2016-07-12         48.0       31.0       Software   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
2604    Problema  2 - Normal        2 - Media              1.0           5.0   
6328   Solicitud  2 - Normal         3 - Alta              7.0           4.0   
1921   Solicitud  2 - Normal         3 - Alta              0.0           5.0   
11265  Solicitud   3 - Mayor  0 - Sin Asignar              3.0           5.0   
489    Solicitud  2 - Normal  0 - Sin Asignar              0.0           5.0   
9213   Solicitud  2 - Normal         3 - Alta              9.0           5.0   
658    Solicitud  2 - Normal  0 - Sin Asignar              0.0           5.0   
378    Solicitud  2 - Normal  0 - Sin Asignar              9.0           5.0   
6358   Solicitud  2 - Normal         3 - Alta              8.0           3.0   
5127    Problema  2 - Normal         3 - Alta              5.0           1.0   

                    web  índice_  
2604   www.tractchun.es      1.0  
6328   www.tractchun.es      1.0  
1921   www.tractchun.es      1.0  
11265  www.tractchun.es      1.0  
489    www.tractchun.es      1.0  
9213   www.tractchun.es      1.0  
658    www.tractchun.es      1.0  
378    www.tractchun.es      1.0  
6358   www.tractchun.es      1.0  
5127   www.tractchun.es      1.0

2 - ¿Cuál es el número de incidencias que se han dado de tipo "Sistema", lo encontraréis en la columna de categoría?

In [5]:
cond1 = df['categoría'] == 'Sistema'
df_sistema = df[cond1]
print(df_sistema.shape)

(5203, 12)


In [6]:
df_sistema.head()
# hay 5203 incidencias del sistema

id_ticket       fecha  id_empleado  id_agente categoría  \
2    GDDESR-1342539995  2016-06-18        569.0       29.0   Sistema   
3    GDDTSR-5942488006  2016-04-28        320.0       40.0   Sistema   
156  GDLESR-0142497241  2016-05-07        584.0        2.0   Sistema   
157  GDLESR-0142558772  2016-07-07       1147.0       47.0   Sistema   
158  GDLESR-0242517083  2016-05-27        578.0       10.0   Sistema   

          tipo           severidad        prioridad  días_resolución  \
2     Problema  0 - Sin Clasificar  0 - Sin Asignar              3.0   
3    Solicitud  0 - Sin Clasificar  0 - Sin Asignar              9.0   
156   Problema          2 - Normal  0 - Sin Asignar              1.0   
157   Problema          2 - Normal  0 - Sin Asignar              1.0   
158   Problema          2 - Normal  0 - Sin Asignar              2.0   

     satisfacción               web  índice_  
2             5.0  www.tractchun.es      1.0  
3             5.0  www.tractchun.es      1.0  
156           5.0  www.tractchun.es      1.0  
157           5.0  www.tractchun.es      1.0  
158           5.0  www.tractchun.es      1.0

3. ¿Y para la categorías de "Hardware" y "Software"? Cread un dataframe nuevo con los resultados de este filtrado de datos. 📌 Nota Este ejercicio se puede resolver usando dos condiciones de filtrado o usando el método isin(). Hacedlo con ambas herramientas.

In [7]:
# metodo isin
df_hs = df[df['categoría'].isin(['Hardware','Software'])]
df_hs.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
2702  GMLTER-1942436677  2016-03-07         97.0       26.0  Software   
2560  GMLEER-2342710941  2016-12-06        924.0       49.0  Software   
5030  KHLEER-0342452181  2016-03-23       1588.0       41.0  Software   
9697  SMLTER-0842522118  2016-06-01       1211.0       11.0  Hardware   
308   GDLTER-3242489601  2016-04-29       1540.0       16.0  Hardware   

           tipo   severidad        prioridad  días_resolución  satisfacción  \
2702  Solicitud  2 - Normal        2 - Media              9.0           5.0   
2560   Problema  2 - Normal        2 - Media              3.0           5.0   
5030   Problema  2 - Normal         3 - Alta              3.0           3.0   
9697  Solicitud  2 - Normal        2 - Media              4.0           4.0   
308   Solicitud  2 - Normal  0 - Sin Asignar              4.0           5.0   

                   web  índice_  
2702  www.tractchun.es      1.0  
2560  www.tractchun.es      1.0  
5030  www.tractchun.es      1.0  
9697  www.tractchun.es      1.0  
308   www.tractchun.es      1.0

In [8]:
cond1 = df['categoría'] == 'Software'
cond2 = df['categoría'] == 'Hardware'
df_hs2 = df[(cond1 | cond2)]
df_hs2.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
4201   KDLTER-7442432410  2016-03-03       1171.0       24.0  Software   
31     GDLEER-4942636742  2016-09-23       1474.0       27.0  Hardware   
2566   GMLEER-3242694605  2016-11-20         20.0       36.0  Software   
3737   GWRTER-3242462260  2016-04-02        106.0       22.0  Software   
10862  TDLTET-4642646630  2016-10-03        923.0       26.0  Software   

            tipo    severidad        prioridad  días_resolución  satisfacción  \
4201   Solicitud   2 - Normal  0 - Sin Asignar              4.0           5.0   
31      Problema   2 - Normal  0 - Sin Asignar              9.0           5.0   
2566    Problema  4 - Crítico        2 - Media              2.0           5.0   
3737   Solicitud    3 - Mayor         1 - Baja             14.0           5.0   
10862  Solicitud   2 - Normal  0 - Sin Asignar              6.0           4.0   

                    web  índice_  
4201   www.tractchun.es      1.0  
31     www.tractchun.es      1.0  
2566   www.tractchun.es      1.0  
3737   www.tractchun.es      1.0  
10862  www.tractchun.es      1.0

In [9]:
cond1 = df['categoría'] == 'Software'
cond2 = df['categoría'] == 'Hardware'
df_hs2 = df[(df['categoría'] == 'Sistema') | (df['categoría'] == 'Hardware')]
df_hs2.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
10416  SWLTSR-2642614667  2016-09-01       1156.0       26.0   Sistema   
12743  TWLTER-2442555076  2016-07-04       1097.0       30.0  Hardware   
246    GDLEST-5442560527  2016-07-09       1002.0       45.0   Sistema   
2988   GMLTSR-5042528368  2016-06-07        856.0       43.0   Sistema   
11196  TDLTSR-9742462889  2016-04-02        348.0        8.0   Sistema   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
10416  Solicitud  2 - Normal         1 - Baja             16.0           1.0   
12743  Solicitud  2 - Normal         1 - Baja              5.0           4.0   
246     Problema  2 - Normal  0 - Sin Asignar              4.0           5.0   
2988   Solicitud  2 - Normal        2 - Media             10.0           4.0   
11196  Solicitud  2 - Normal  0 - Sin Asignar              7.0           4.0   

                    web  índice_  
10416  www.tractchun.es      1.0  
12743  www.tractchun.es      1.0  
246    www.tractchun.es      1.0  
2988   www.tractchun.es      1.0  
11196  www.tractchun.es      1.0

- 3.1 De los datos filtrados en el punto 3. ¿Cuántas de ellas son de tipo "Solicitud" y cuántas de tipo "Problema"?

In [10]:
df_hs['categoría'].value_counts()

Software    2576
Hardware    1254
Name: categoría, dtype: int64

- 3.2 ¿Cuántas solicitudes tienen una satisfacción de más de 3?

In [11]:
df_hssatis = df_hs[df_hs['satisfacción'] > 3]
df_hssatis.shape
# hay 2995

(2995, 12)

- 3.3 ¿Y cuántas tienen una satisfacción mayor o igual que 4 y han tardado menos de 3 días de resolución?

In [12]:
cond_a = df_hs['satisfacción'] >= 4
cond_b = df_hs['días_resolución'] < 3

In [13]:
df_satisresol = df_hs[(cond_a & cond_b)]
df_satisresol.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
1580   GHLTER-8242464151  2016-04-04       1505.0       41.0  Hardware   
1556   GHLTER-7242573711  2016-07-22        831.0       27.0  Software   
7822   SDLEER-3242614075  2016-09-01       1897.0       10.0  Software   
3177   GWLEET-0242722047  2016-12-18       1744.0       10.0  Software   
11324  THLEER-5042433800  2016-03-04        200.0       18.0  Software   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
1580   Solicitud  2 - Normal         3 - Alta              0.0           5.0   
1556   Solicitud  2 - Normal         3 - Alta              2.0           5.0   
7822    Problema  2 - Normal  0 - Sin Asignar              2.0           5.0   
3177    Problema  2 - Normal         1 - Baja              2.0           5.0   
11324   Problema  2 - Normal         3 - Alta              0.0           4.0   

                    web  índice_  
1580   www.tractchun.es      1.0  
1556   www.tractchun.es      1.0  
7822   www.tractchun.es      1.0  
3177   www.tractchun.es      1.0  
11324  www.tractchun.es      1.0

In [14]:
df_satisresol.shape
# hay 574

(574, 12)

4 - Ahora nuestro jefe quiere saber que cantidad de tickets ha habido en la segunda quincena de Enero. Cread un DataFrame nuevo con los tickets correspondientes a la segunda quincena de Enero.

In [15]:
df_hs

id_ticket       fecha  id_empleado  id_agente categoría  \
4      GDLEER-0042524120  2016-06-03       1842.0       31.0  Software   
5      GDLEER-0142608095  2016-08-26         59.0       20.0  Software   
6      GDLEER-0242564650  2016-07-13       1175.0       36.0  Software   
7      GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8      GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
...                  ...         ...          ...        ...       ...   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0  Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0  Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0  Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0  Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
4       Problema  2 - Normal  0 - Sin Asignar              0.0           5.0   
5       Problema  2 - Normal  0 - Sin Asignar              1.0           1.0   
6       Problema  2 - Normal  0 - Sin Asignar              2.0           1.0   
7       Problema  2 - Normal  0 - Sin Asignar              5.0           5.0   
8       Problema  2 - Normal  0 - Sin Asignar              8.0           5.0   
...          ...         ...              ...              ...           ...   
13034  Solicitud   1 - Menor         1 - Baja             10.0           4.0   
13035  Solicitud   1 - Menor         1 - Baja             10.0           4.0   
13036  Solicitud   1 - Menor         1 - Baja              7.0           5.0   
13037  Solicitud   1 - Menor         1 - Baja              5.0           5.0   
13038  Solicitud   3 - Mayor         1 - Baja              7.0           5.0   

                    web  índice_  
4      www.tractchun.es      1.0  
5      www.tractchun.es      1.0  
6      www.tractchun.es      1.0  
7      www.tractchun.es      1.0  
8      www.tractchun.es      1.0  
...                 ...      ...  
13034  www.tractchun.es      1.0  
13035  www.tractchun.es      1.0  
13036  www.tractchun.es      1.0  
13037  www.tractchun.es      1.0  
13038  www.tractchun.es      1.0  

[3830 rows x 12 columns]

In [21]:
patron = '2016-01-(1[5-9]|[2-3][0-9])'

In [22]:
df_enero = df_hs[df_hs['fecha'].str.contains(patron)]
df_enero.sample(5)

/tmp/ipykernel_1961/510028909.py:1: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df_enero = df_hs[df_hs['fecha'].str.contains(patron)]


id_ticket       fecha  id_empleado  id_agente categoría  \
10779  TDLTER-4942400399  2016-01-31       1439.0       33.0  Software   
10870  TDLTET-7342384993  2016-01-15       1889.0        9.0  Software   
9746   SMLTER-4742393561  2016-01-24        516.0       35.0  Software   
9696   SMLTER-0742387296  2016-01-18        149.0       22.0  Software   
4179   KDLTER-6542393286  2016-01-24        248.0       12.0  Software   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
10779  Solicitud  2 - Normal  0 - Sin Asignar              9.0           1.0   
10870  Solicitud  2 - Normal  0 - Sin Asignar              3.0           5.0   
9746   Solicitud  2 - Normal        2 - Media              7.0           4.0   
9696   Solicitud  2 - Normal        2 - Media              7.0           1.0   
4179   Solicitud  2 - Normal  0 - Sin Asignar              5.0           4.0   

                    web  índice_  
10779  www.tractchun.es      1.0  
10870  www.tractchun.es      1.0  
9746   www.tractchun.es      1.0  
9696   www.tractchun.es      1.0  
4179   www.tractchun.es      1.0

- 4.1 ¿Ha habido algun ticket en esta quincena que corresponda a la categoría de Solicitud?

In [25]:
df_enero[df_enero['tipo'] == 'Solicitud'].shape
# 155

(155, 12)

- 4.2 Revisando los datos, nuestro jefe se ha dado cuenta que hay muchos tickets no tienen ninguna prioridad asignada. Esto no debería seguir repitiéndose ya que cada ticket debe tener una prioridad para saber que urgencia debemos darle a cada ticket. 

- Usando el **str.contains** extraed los tickets que no tengan una prioridad asignada o que tengan una prioridad baja.

In [30]:
patron2 = '^[0]|^[1]'

In [33]:
df_prioridad = df[df['prioridad'].str.contains(patron2)]

In [34]:
df_prioridad.sample(5)

id_ticket       fecha  id_empleado  id_agente      categoría  \
12835  TWLTNR-3142431646  2016-03-02        724.0        6.0  Acceso&%Login   
3151   GWLEER-2542570457  2016-07-19         65.0       24.0       Hardware   
7942   SDLENT-8042642826  2016-09-29       1082.0       28.0  Acceso&%Login   
433    GDLTET-0942413979  2016-02-13        677.0        9.0       Hardware   
10488  SWLTSR-8442604249  2016-08-22        834.0       22.0        Sistema   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
12835  Solicitud  2 - Normal         1 - Baja              1.0           4.0   
3151    Problema  2 - Normal         1 - Baja             15.0           5.0   
7942    Problema  2 - Normal  0 - Sin Asignar              0.0           1.0   
433    Solicitud  2 - Normal  0 - Sin Asignar              5.0           5.0   
10488  Solicitud  2 - Normal         1 - Baja             14.0           4.0   

                    web  índice_  
12835  www.tractchun.es      1.0  
3151   www.tractchun.es      1.0  
7942   www.tractchun.es      1.0  
433    www.tractchun.es      1.0  
10488  www.tractchun.es      1.0

5 - Nuestro jefe quiere seguir buscando patrones y quiere saber si existe una relación entre el número de días que tarde en cerrarse un ticket y la satisfacción de los clientes. Para eso nos pide que:

- 5.1 Extraigamos los tickets en lo que pasaron 5 o más días en ser solucionados.

In [38]:
df_resolucion_larga = df[df['días_resolución'] >= 5]

In [39]:
df_resolucion_larga.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
9451   SHLTST-5942398197  2016-01-29        419.0       41.0   Sistema   
11622  THLTER-9542588240  2016-08-06       1994.0       12.0  Software   
4146   KDLTER-5042703548  2016-11-29       1844.0       25.0  Software   
927    GDLTSR-8742658224  2016-10-15       1942.0        2.0   Sistema   
7731   KWLTSR-7942632190  2016-09-19        879.0       21.0   Sistema   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
9451   Solicitud  2 - Normal         3 - Alta              9.0           5.0   
11622  Solicitud  2 - Normal         3 - Alta              5.0           4.0   
4146   Solicitud  2 - Normal  0 - Sin Asignar             10.0           1.0   
927    Solicitud  2 - Normal  0 - Sin Asignar              7.0           5.0   
7731   Solicitud  2 - Normal         1 - Baja              9.0           5.0   

                    web  índice_  
9451   www.tractchun.es      1.0  
11622  www.tractchun.es      1.0  
4146   www.tractchun.es      1.0  
927    www.tractchun.es      1.0  
7731   www.tractchun.es      1.0

- 5.2 Extraigamos los tickets en los que pasaron menos de 5 días en ser solucionados.

In [40]:
df_resolucion_corta = df[df['días_resolución'] < 5]

In [41]:
df_resolucion_corta.sample(5)

id_ticket       fecha  id_empleado  id_agente categoría  \
11470  THLESR-8442427177  2016-02-27       1547.0       41.0   Sistema   
2575   GMLEER-7042730232  2016-12-26       1953.0       22.0  Software   
462    GDLTET-5342424587  2016-02-24       1348.0       15.0  Software   
3125   GMRTSR-2942390754  2016-01-21        805.0       37.0   Sistema   
4257   KDLTET-0342686232  2016-11-12       1523.0       12.0  Software   

            tipo    severidad        prioridad  días_resolución  satisfacción  \
11470   Problema   2 - Normal         3 - Alta              4.0           4.0   
2575    Problema  4 - Crítico        2 - Media              1.0           1.0   
462    Solicitud   2 - Normal  0 - Sin Asignar              3.0           5.0   
3125   Solicitud    3 - Mayor        2 - Media              1.0           5.0   
4257   Solicitud   2 - Normal  0 - Sin Asignar              3.0           1.0   

                    web  índice_  
11470  www.tractchun.es      1.0  
2575   www.tractchun.es      1.0  
462    www.tractchun.es      1.0  
3125   www.tractchun.es      1.0  
4257   www.tractchun.es      1.0

- 5.3 Usando como referencia los dos conjuntos de datos creados en los dos pasos anteriores:

- ¿Qué valores tenemos de satisfacción en cada uno de ellos?

In [44]:
df_resolucion_larga['satisfacción'].value_counts()

5.0    2873
4.0    1803
1.0     718
3.0     533
2.0     109
Name: satisfacción, dtype: int64

In [45]:
df_resolucion_corta['satisfacción'].value_counts()

5.0    3314
4.0    1926
1.0     865
3.0     593
2.0     167
Name: satisfacción, dtype: int64

- ¿En cuál de ellos hay más puntuaciones de 5 en relación a la satisfacción?

En la tabla en la que los dias de resolucion es más corta hay más puntuaciones de satisfaccion 5 (3314)
Frente a la tabla en la que se encuentran los tikets con un tiempo de resolucion más largo (2873)

6 - Revisando los dataframes generados en el ejercicio anterior, nos hemos dado cuenta que hay una serie de empleados que tardan demasiado en resolver los problemas planteados por los clientes. Nuestro jefe quiere extraer los datos de estos empleados en concreto para saber que esta pasando. 

- En concreto estos empleados son aquellos que tienen los siguientes id's: 1118.0, 730.0, 97.0, 1745.0. Crea un dataframe donde solo tengamos la información de estos empleados. Usad el método isin(), para resolver este ejercicio.

In [54]:
df_empleadoslentos = df[df['id_empleado'].isin([1118.0, 730.0, 97.0, 1745.0])]

In [56]:
df_empleadoslentos.sample(3)

id_ticket       fecha  id_empleado  id_agente categoría  \
8605  SHLEER-0442515806  2016-05-25        730.0       28.0  Software   
7090  KMLTSR-9542558851  2016-07-07       1745.0       48.0   Sistema   
5985  KHLTSR-0642569057  2016-07-18       1745.0       10.0   Sistema   

           tipo   severidad  prioridad  días_resolución  satisfacción  \
8605   Problema  2 - Normal   3 - Alta              4.0           5.0   
7090  Solicitud  2 - Normal  2 - Media              5.0           5.0   
5985  Solicitud  2 - Normal   3 - Alta              6.0           3.0   

                   web  índice_  
8605  www.tractchun.es      1.0  
7090  www.tractchun.es      1.0  
5985  www.tractchun.es      1.0

7 - Utilizando la columna de fecha cread una nueva columna llamada semestre que contenga 1er semestre o 2º semestre. 
- Aquellos tickets que pertenezcan a los seis primeros meses tendran el valor de "1er trismestre" y los tickets de julio en adelante tendrán el valor de "2º semestre".

In [57]:
df.sample(3)

id_ticket       fecha  id_empleado  id_agente categoría  \
10846  TDLTER-9442553936  2016-07-02        453.0       19.0  Software   
11080  TDLTSR-2742675572  2016-11-01       1117.0       25.0   Sistema   
2689   GMLTER-0942544958  2016-06-23         35.0       19.0  Software   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
10846  Solicitud  2 - Normal  0 - Sin Asignar              4.0           4.0   
11080  Solicitud  2 - Normal  0 - Sin Asignar             17.0           5.0   
2689   Solicitud  2 - Normal        2 - Media              9.0           1.0   

                    web  índice_  
10846  www.tractchun.es      1.0  
11080  www.tractchun.es      1.0  
2689   www.tractchun.es      1.0

In [58]:
p_semestre1 = '2016-0[1-6].*'           
p_semestre2 = '2016-0[7-9]|1[0-2].*' 

In [59]:
condicion1 = df['fecha'].str.contains('2016-0[1-6].*')
condicion2 = df['fecha'].str.contains('2016-0[7-9]|1[0-2].*')

In [60]:
opciones = ['1º semestre', '2º semestre']

In [62]:
df['semestre'] = np.select([condicion1,condicion2], opciones)

In [63]:
df.sample(3)

id_ticket       fecha  id_empleado  id_agente categoría  \
862    GDLTSR-5842701823  2016-11-27       1322.0       18.0   Sistema   
11335  THLEER-7442528968  2016-06-07       1366.0       19.0  Software   
11585  THLTER-7442554452  2016-07-03       1315.0       14.0  Software   

            tipo   severidad        prioridad  días_resolución  satisfacción  \
862    Solicitud  2 - Normal  0 - Sin Asignar             14.0           1.0   
11335   Problema  2 - Normal         3 - Alta              4.0           4.0   
11585  Solicitud  2 - Normal         3 - Alta              4.0           4.0   

                    web  índice_     semestre  
862    www.tractchun.es      1.0  2º semestre  
11335  www.tractchun.es      1.0  1º semestre  
11585  www.tractchun.es      1.0  2º semestre

8 - Guardad como un archivo csv nuevo. Lo usaremos en el pair programming de mañana.

In [65]:
df.to_csv("datos/df_tikets.csv")